### PR TITLE
Docs: add guide for generic platforms

### PR DESCRIPTION
**Pull request by robszumski**
Supersedes https://github.com/coreos/coreos-kubernetes/pull/618

cc: @pbx0 @aaronlevy

### CODE REVIEW

File: multi-node/generic/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Kubernetes on CoreOS with Generic Install Scripts
+
+TThis guide will setup Kubernetes on CoreOS in a similar way to other tools in the repo. The main goal of these scripts is to be generic and work on many different cloud providers or platforms. The notable difference is that these scripts are intended to be platform agnostic and thus don't automatically setup the TLS assets on each host beforehand.
```
Comment (pbx0) [inline multi-node/generic/README.md]: s/TThis/This

**Comment by pbx0** [comment]
Thanks for redoing this, looks good to me overall. What do you think about leaving in a pointer to coreos-baremetal?

**Comment by robszumski** [comment]
@pbx0 I added back in your paragraph about vagrant testing and coreos-baremetal. I think its a nice point.

**Comment by pbx0** [comment]
LGTM

**Comment by robszumski** [comment]
Squashed/rebased

**Pull request merged**